engine/engine.go

```diff
@@ -65,6 +65,10 @@ type Opts struct {
 	// If nil, nothing will be printed.
 	// NOTE: Users will not check the errors, debug writing is best effort.
 	DebugWriter io.Writer
+
+	// ExtLookbackDelta specifies what time range to use to determine valid previous sample for extended range functions.
+	// Defaults to 1 day if not specified.
+	ExtLookbackDelta time.Duration
```

Comment:
This might be too high as a default because the function will buffer 1 day of data for each series. Maybe we could reduce it to 1h and we can still extend it by explicitly setting the parameter.